src/Typechecker/Typechecker.hs

```diff
@@ -38,9 +48,13 @@ deftype (Val (Sig n t) eqn x) = do
 
 deftype (BVal (Sig n t) eqs x) = do
   setPos x
+  (mx, my) <- getSize
+  -- get a set of the spaces covered by these board equations
+  let placesCovered = S.fromList $ concat $ map (getCovered (mx,my)) eqs
   eqTypes <- mapM beqntype eqs
   case all' (<= t) eqTypes of
-    Nothing -> return t
+    -- pass if all spaces are defined, otherwise incomplete/uninitialized
+    Nothing -> if length placesCovered == mx*my then return t else uninitialized n
```

Comment:
It would be better to do an index-wise comparison rather than merely checking for length. Generate the set of cells from the size and check for equality on that.

Reply:
So maybe we can talk about this today, since there's a couple of ways to do this. I was thinking about what would work, since we have multiple equations that can cover places, some covering the same places, I figured it was easier just to compute a list of positions that would be covered by all equations, and then generate a set from that. The typechecking is already performed for the result of each equation, so all spaces that are set should be typechecked correctly at this point.